### PR TITLE
chore(suse): update EOL dates for SUSE SLES 15.3, 15.4 and OpenSUSE 15.4

### PR DESCRIPTION
**Pull request by macedogm**
## Description
SUSE SLES 15.4 and OpenSUSE 15.4 were released, so this PR updates its EOL dates and also for SLES 15.3.

## Related issues
- Close https://github.com/aquasecurity/trivy/issues/2402

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

Signed-off-by: Guilherme Macedo <guilherme.macedo@suse.com>

### CODE REVIEW

**Comment by macedogm** [comment]
CCing @dirkmueller

File: pkg/detector/ospkg/suse/suse.go
Comment (dirkmueller) [inline pkg/detector/ospkg/suse/suse.go:42]: can you keep / readd the comment `6 months after SLES 15 SP5 release` here?

File: pkg/detector/ospkg/suse/suse.go
Comment (dirkmueller) [inline pkg/detector/ospkg/suse/suse.go:41]: no, its 12, 31. see https://www.suse.com/lifecycle/

image

**Review by dirkmueller — CHANGES_REQUESTED** [review]
please update the end date for 15.3

**Comment by macedogm** [comment]
Fixed.

**Comment by knqyf263** [comment]
@macedogm Thanks! Please let me wait for approval from @dirkmueller.

BTW, is there any way to get these EOL dates via API or something like that? We'd like to automate it.

**Comment by macedogm** [comment]
> BTW, is there any way to get these EOL dates via API or something like that? We'd like to automate it.

I'll defer this question to @dirkmueller.

**Review by dirkmueller — APPROVED** [review]

**Comment by dirkmueller** [comment]
> BTW, is there any way to get these EOL dates via API or something like that? We'd like to automate it.

I believe we currently don't offer it via a public API. I will check this and file an issue/PR

**Review by knqyf263 — APPROVED** [review]

**Comment by knqyf263** [comment]
@dirkmueller Thanks.

**Pull request merged**
